castle/cms/caching/fastly.py

```diff
@@ -0,0 +1,32 @@
+from castle.cms.caching.purgemanager import PurgeManager
+from plone.registry.interfaces import IRegistry
+from zope.component import getUtility
+
+import requests
+
+
+class Fastly(PurgeManager):
+    def __init__(self):
+        super(Fastly, self).__init__()
+        registry = getUtility(IRegistry)
+        self.fastly_token = registry.get('castle.fastly_token', None)
+        self.fastly_service_id = registry.get('castle.fastly_service_id', None)
+        self.enabled = (
+            self.fastly_token is not None and
+            self.fastly_service_id is not None)
+
+    def getUrlsToPurge(self, path):
+        return super(Fastly, self).getUrlsToPurge(path)
+
+    def purge(self, urls):
+        url = "https://api.fastly.com/service/%s/purge_all" % self.fastly_service_id
```

Comment:
Ideally, we don't purge everything from the fastly cache, just the specific url's we get from getUrlsToPurge() -- does fastly have no interface to purge individual files/urls?